lib/cstruct.ml

```diff
@@ -95,6 +95,12 @@ let create len =
 let check_bounds t len =
   Bigarray.Array1.dim t.buffer >= len
 
+external address_bigstring : buffer -> int64 = "caml_address_bigstring"
+
+let check_alignment t alignment =
+  let address = address_bigstring t.buffer in
+  Int64.(rem (add address (of_int t.off)) (of_int alignment) = 0L)
```

Comment:
There are an awful lot of minor heap allocations here... if this is in the fast path as an assert it will probably increase GC pressure